worker/uniter/runner/context/context.go

```diff
@@ -637,6 +645,11 @@ func (context *HookContext) HookVars(paths Paths) ([]string, error) {
 		"JUJU_PRINCIPAL_UNIT="+context.principal,
 		"JUJU_AVAILABILITY_ZONE="+context.availabilityzone,
 		"JUJU_VERSION="+version.Current.String(),
+		// Some of these will be empty, but that is fine, better
+		// to explicitly export them as empty.
+		"JUJU_CHARM_HTTP_PROXY="+context.proxySettings.Http,
+		"JUJU_CHARM_HTTPS_PROXY="+context.proxySettings.Https,
+		"JUJU_CHARM_FTP_PROXY="+context.proxySettings.Ftp,
```

Comment:
Could add JUJU_CHARM_SOCKS_PROXY as well - at least for charm usage.

To use a SOCKS proxy a charm would only need to use a URL, e.g.:
https://www.chromium.org/developers/design-documents/network-stack/socks-proxy
socks5://myproxy:1080

golang has socks5 proxy handling too for HTTP transport and uses "socks5" as a scheme:
https://github.com/golang/go/blob/release-branch.go1.10/src/net/http/transport.go#L296

Reply:
When we have a socks proxy setting, yes. This branch just exposes the ones we current have with env vars that shouldn't change as we move forwards.

Reply:
Ack, agreed.